src/plugins/intel_cpu/src/nodes/kernels/scaled_attn/softmax_kernel.hpp

```diff
@@ -432,22 +484,42 @@ inline void exp_reduce_sum(float* a, const float max, const size_t size, float&
         v_a = _mm256_blendv_ps(_mm256_setzero_ps(), v_a, _mm256_castsi256_ps(mask));
         v_sum = _mm256_add_ps(v_a, v_sum);
         _mm256_maskstore_ps(a + i, mask, v_a);
+
+        i += (size - i);
     }
     hsum(v_sum);
     sum = _mm256_cvtss_f32(v_sum);
-#else
-    for (size_t i = 0; i < size; i++) {
+#elif defined(OPENVINO_ARCH_ARM64)
+    float32x4_t v_a;
+    float32x4_t v_max = vdupq_n_f32(max);
+    float32x4_t v_sum = vdupq_n_f32(0.0f);
+
+    while (i + vec_len_f32_neon <= size) {
+        v_a = vld1q_f32(a + i);
+        v_a = vsubq_f32(v_a, v_max);
+        v_a = exp_ps_neon(v_a);
+        v_sum = vaddq_f32(v_sum, v_a);
+        vst1q_f32(a + i, v_a);
+        i += vec_len_f32_neon;
+    }
+    float32x2_t vlow = vget_low_f32(v_sum);
+    float32x2_t vhigh = vget_high_f32(v_sum);
+    vlow = vpadd_f32(vlow, vhigh);
+    vlow = vpadd_f32(vlow, vlow);
```

Comment:
1. lets extract into 'hsum' func and put into common.hpp
2. Have you tried 'vaddvq_f32'? Seems like the same math can be done in one instruction call